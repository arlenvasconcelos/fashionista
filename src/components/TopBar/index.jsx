import Topbar from './TopBar'

export default Topbar