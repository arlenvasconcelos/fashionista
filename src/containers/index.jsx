import App from './App/App';
import Products from './Products/Products';
import Cart from './Cart/Cart';
import SingleProduct from './SingleProduct/SingleProduct';

export {App, Products, Cart, SingleProduct}