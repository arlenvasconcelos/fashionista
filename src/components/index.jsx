import CartDetails from './CartDetails/CartDetails';
import CartFooter from './CartFooter/CartFooter';
import CartHeader from './CartHeader/CartHeader';
import Drawer from './Drawer/Drawer';
import Loading from './Loading/Loading';
import NavSearch from './NavSearch/NavSearch';
import NoImage from './NoImage/NoImage';
import Notification from './Notification/Notification';
import Product from './Product/Product';
import ProductDetails from './ProductDetails/ProductDetails';
import ProductImage from './ProductImage/ProductImage';
import ProductPrice from './ProductPrice/ProductPrice';
import TopBar from './TopBar/TopBar';

export {
  CartDetails, CartFooter, CartHeader, Drawer, Loading,
  NavSearch, NoImage, Notification, Product, ProductDetails, 
  ProductImage, ProductPrice, TopBar,
}