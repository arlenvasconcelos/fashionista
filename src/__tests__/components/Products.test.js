import React from "react";
import { mount, shallow, render } from "enzyme";
import { Provider } from "react-redux";
import { BrowserRouter as Router, Route } from "react-router-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import createStore from "redux-mock-store";

import { Products } from "../../containers";
import { Product } from "../../components";

Enzyme.configure({ adapter: new Adapter() });

const product = {
  actual_price: "R$ 199,90",
  code_color: "20002605_613",
  color: "TAPEÃ‡ARIA",
  color_slug: "tapecaria",
  discount_percentage: "",
  image:
    "https://viniciusvinna.netlify.app/assets/api-fashionista/20002605_615_catalog_1.jpg",
  installments: "3x R$ 66,63",
  name: "VESTIDO TRANSPASSE BOW",
  on_sale: false,
  regular_price: "R$ 199,90",
  sizes: [
    { available: false, size: "PP", sku: "5807_343_0_PP" },
    { available: true, size: "P", sku: "5807_343_0_P" },
    { available: true, size: "M", sku: "5807_343_0_M" },
    { available: true, size: "G", sku: "5807_343_0_G" },
    { available: false, size: "GG", sku: "5807_343_0_GG" },
  ],
};

describe("Products Test", () => {
  const mockStore = createStore();
  const INITIAL_STATE = {
    products: {
      allProducts: [],
      filteredProducts: [],
      openedFilter: false,
      errorStatus: false,
      errorMessage: "Error message",
    },
    cart: {
      items: [],
      open: false,
    },
  };

  const store = mockStore(INITIAL_STATE);
  const wrapper = mount(
    <Router>
      <Provider store={store}>
        <Route>
          <Products products={[product]} />
        </Route>
      </Provider>
    </Router>
  );

  it("should match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("should render Products", () => {
    expect(wrapper.find(".products")).toBeTruthy();
  });

  it("should find products__noproduct", () => {
    expect(wrapper.find(".products__noproduct")).toBeTruthy();
  });

  it("should find products__quantity", () => {
    expect(wrapper.find(".products__quantity")).toBeTruthy();
    expect(wrapper.find(".products__quantity").text()).toContain("1");
  });

  it("should find product", () => {
    expect(wrapper.find(Product)).toHaveLength(1);
  });
});
