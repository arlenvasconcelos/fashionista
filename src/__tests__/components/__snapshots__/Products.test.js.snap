// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Test should match the snapshot 1`] = `"<div class=\\"container\\"><main class=\\"products\\"><div class=\\"products__quantity\\"><p>1 items</p></div><div class=\\"products__items\\"><section class=\\"product\\" data-testid=\\"product\\"><a href=\\"/produto/20002605_613\\"><div class=\\"product__content\\"><img src=\\"https://viniciusvinna.netlify.app/assets/api-fashionista/20002605_615_catalog_1.jpg\\" alt=\\"VESTIDO TRANSPASSE BOW\\"><div class=\\"product__footer\\"><div class=\\"product__name\\">VESTIDO TRANSPASSE BOW</div><div class=\\"product__price\\">R$ 199,90</div></div></div></a></section></div><div class=\\"navsearch \\"><div class=\\"navsearch__icon\\"><button><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"><circle cx=\\"11\\" cy=\\"11\\" r=\\"8\\"></circle><line x1=\\"21\\" y1=\\"21\\" x2=\\"16.65\\" y2=\\"16.65\\"></line></svg></button></div></div></main></div>"`;
